README.md

```diff
@@ -0,0 +1,203 @@
+the repository root. The layout defines the steps `target-develop`,
+`target-code-review`, `target-get-dependency`, and `target-jfrog-xray` which
+handle code development, code review (a manual process), the import of a package
+`dependency`, and a scan for vulnerbailities using JFrog Xray (mocked for this
+demo).
+
+__Dependency Supply Chain__
+
+The script will create a `root.layout` file in the `metadata_dependency` folder
+in the repository root. The layout defines the steps `dependency-develop` and
+`dependency-code-review`. For the purposes of this demo, we're assuming the
+developer and reviewer are the same person for both supply chains.
+
+### Develop
+
+This process is carried out by the developer. There is no universal command or
+editor for developers to create their programs, so this step has no
+`expected_command`. Instead, it expects developers to write code and generate
+link metadata for the `products` they created.
+
+```bash
+cd ../developer
+in-toto-run --step-name target-develop --products ../target/demo.py --key developer --no-command
+in-toto-run --step-name dependency-develop --products ../dependency/demo.py --key developer --no-command
+```
+
+This generates signed link metadata in the developer's folder, which we copy to
+the metadata folder in the project root.
+
+```bash
+mv target-develop.7cedd7de.link ../metadata_target/
+mv dependency-develop.7cedd7de.link ../metadata_dependency/
+```
+
+### Code Review
+
+This process again has no universal command - every team and organization has
+their own code review process. Therefore, the functionary for this step merely
+generates link metadata indicating that they've signed off on the code review.
+
+```bash
+cd ../reviewer
+in-toto-run --step-name target-code-review --materials ../target/demo.py --products ../target/demo.py --key reviewer --no-command
+in-toto-run --step-name dependency-code-review --materials ../dependency/demo.py --products ../dependency/demo.py --key reviewer --no-command
+```
+
+This generates signed link metadata in the reviewer's folder, which we copy to
+the metadata folder in the project root.
+
+```bash
+mv target-code-review.3b5fef71.link ../metadata_target/
+mv dependency-code-review.3b5fef71.link ../metadata_dependency/
+```
+
+### Scanning with JFrog Xray (mocked)
+
+This process is expected to make use of the REST APIs provided as part of the
+XRay service. However, for , it's a simple script to generate a report similar
+to the ones generated by XRay.
+
+```bash
+cd ../xray
+in-toto-run --step-name target-jfrog-xray --key xray --products ../reports/jfrog-xray-report.json --materials ../target -- python generate_report.py
+```
+
+The script generates a JSON file called `jfrog-xray-report.json` in the reports
+folder in the project root. The step generates signed link metadata in the
+xray folder, which we copy to the metadata folder in the project root.
+
+```bash
+mv target-jfrog-xray.8cf1c499.link ../metadata_target/
+```
```

Comment:
wouldn't it be easier to embed the report json inside the link using the `env` key? Then the verifier can load the piece of link.